java-checks/src/main/java/org/sonar/java/checks/RedundantTypeCastCheck.java

```diff
@@ -39,14 +46,11 @@
 import org.sonar.plugins.java.api.tree.TypeCastTree;
 import org.sonar.plugins.java.api.tree.VariableTree;
 
-import javax.annotation.CheckForNull;
-
-import java.util.List;
-import java.util.Set;
-
 @Rule(key = "S1905")
 public class RedundantTypeCastCheck extends IssuableSubscriptionVisitor {
 
+  private static final Predicate<JavaSymbol> ABSTRACT_METHOD_PREDICATE = symbol -> !symbol.isDefault();
```

Comment:
Dodgy name : this is just about default not only about abstract. You can have a method with a body which is not default in class that are not interfaces.